src/main/java/org/gradle/profiler/ScenarioInvoker.java

```diff
@@ -74,10 +76,30 @@ protected void runCleanup(BuildContext buildContext, BuildMutator mutator, Runna
                 new CommandExec().inDir(workingDir).runAndCollectOutput(buildLog, commandLine);
             }
             Duration executionTime = timer.elapsed();
-            return new BuildInvocationResult(buildContext, executionTime);
+            return (T) new BuildInvocationResult(buildContext, executionTime);
         };
     }
 
+    void doRun(T scenario, InvocationSettings settings, Consumer<BuildInvocationResult> resultConsumer, List<String> commandLine) {
```

Comment:
This method looks like it shouldn't be here, since it is not used by the `GradleScenarioInoker`. Maybe there should be an intermediate class for a `CommandLineBuildToolScenarioInvoker`?

I also saw that the `CliInvoker` (Gradle) uses `ProcessBuilder` directly while `measureCommandLineInvocation` uses `CommandExec`.

Reply:
Done.